django/core/handlers/middleware.py

```diff
@@ -0,0 +1,10 @@
+class MiddlewareMixin(object):
+    def __call__(self, request, response_factory):
+        response = None
+        if hasattr(self, 'process_request'):
+            response = self.process_request(request)
+        if not response:
+            response = response_factory(request)
```

Comment:
No, since the existing code performs like that -- for better or worse (`response_factory` will call it internally with the same semantics as currently, unless I mixd that up).